### ISSUE TITLE
how to draw a sample path from a prior GP? (None inputs not supported)

### ISSUE THREAD

**Issue by shalijiang** [pytorch/botorch#157]
## Issue description

I'd like to do some simulation experiments where the function is drawn from a GP, but it seems that current implementation of, e.g., SingleTaskGP, doesn't support passing in None and train_inputs or train_targets as the original ExactGP does. We could use train mode to ignore the training inputs, but GPyTorch forces checking the inputs so that they are consistent with the model.train_inputs by hardcoding settings.debug.on = True. 

Is there anyother way to draw from a SingleTaskGP prior?

**Comment by Balandat** [comment]
Thanks for raising this, we should indeed support this better.

For now, one way of doing this is to use train mode (as you suggested), but disabling the debug setting:
```
with gpytorch.settings.debug(False):
    gp.train()
    out = gp(test_x)
    gp.eval()
```

Another hacky way that requires you to remove the train inputs temporarily is just to set these to `None`:
```
train_inputs = gp.train_inputs
gp.train_inputs = None
out = gp(test_x)
gp.train_inputs = train_inputs
```

Let me know if this helps. 


What we could also do on the gpytorch side is add a setting for prior mode that can be used with a context manager for any model. That is, we'd check for that setting [here](https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/models/exact_gp.py#L233), and then you could call any model (training data or not) in prior mode by doing
```
with gpytorch.settings.prior_mode(True):
    out = gp(test_x)
```
@jacobrgardner, @gpleiss  any thoughts on this?

**Comment by gpleiss** [comment]
That sounds like a good idea. I'm opening up an issue in gpytorch about that: https://github.com/cornellius-gp/gpytorch/issues/696

**Comment by Balandat** [comment]
@shalijiang, I implemented this in https://github.com/cornellius-gp/gpytorch/pull/707.

Closing out this issue as I'm assuming it's resolved once that gpytorch change goes in. Feel free to reopen if that's not the case.
